x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/LocateFunctionProcessorDefinition.java

```diff
@@ -85,6 +85,10 @@ public final void collectFields(SqlSourceBuilder sourceBuilder) {
 
     @Override
     protected NodeInfo<LocateFunctionProcessorDefinition> info() {
+        if (start == null) {
+            return NodeInfo.create(this, LocateFunctionProcessorDefinition::new, expression(), pattern, source);
```

Comment:
Hmmmm. I think it might be less error prone if only have the one constructor for LocateFunctionProcessorDefinition. Is that possible?

Reply:
@nik9000 I know I already tried this as a first option when I started implementing these, but I don't remember why it didn't work out. I _think_ it was about the same reason: null children in the constructor. I'll revisit, maybe I'll find another way.

Reply:
If that funny test gets in your way then ping me! I think I can help work around it.....